docs/backend/api-definitions-guidelines.md

```diff
@@ -0,0 +1,20 @@
+# API Definitions Guidelines
+
+This document describes how service APIs will be managed for `InstructLab` and the sub-components of the `InstructLab` backend.
+
+## What parts of InstructLab need service APIs?
+
+There are two primary classes of service APIs needed to support InstructLab:
+
+* `Platform APIs`: These are APIs that are ignorant of `InstructLab` and provide generic AI platform capabilities (e.g., Fine Tuning, SDG, Eval)
+* `InstructLab APIs`: These are the APIs that reflect the user-facing functionality of `InstructLab` itself. They are aware of the end-to-end `InstructLab` workflow.
+
+The `InstructLab APIs` are essential for hosting `InstructLab` as a service in a repeatable way. The `Platform APIs` are critical for component reuse and extensibility (e.g., new SDG algorithms for new taxonomy data types), but they are not strictly required for hosting `InstructLab` as a service.
+
+## How will service APIs be defined?
+
+Service APIs will be defined using [OpenAPI](https://www.openapis.org/) format in [YAML](https://yaml.org/). For structural and style guidelines, see [api-definitions](../../api-definitions/README.md).
+
+## Where will service API definitions live?
```

Comment:
How does the project guarantee that (manually written?) specs correspond to implementations? Wonder if these should be automatically generated by the REST framework used and not manually maintained separately from implementations.

Is there compliance testing planned for these APIs?

Reply:
I think the point of using openapi to describe the API is also to use side-tooling to auto-generate clients and SDKs, even server-side.

Reply:
Yes, agreed. The goal of the OpenAPI specs is to leverage codegen and/or validation as much as possible in the server implementation language of choice. Per [the conversation below](https://github.com/instructlab/dev-docs/pull/71/files#r1630747990), I think this is a good direction to consider where we have an isolated repo with the specs that creates distributed packages for our supported language.

Reply:
I've. now refactored this PR to propose putting the definitions themselves into their own repo. Unfortunately, this makes it a little hard to find the conversation that led to the change!